src/api/telemetry/TelemetryCollection.js

```diff
@@ -92,6 +92,8 @@ export default class TelemetryCollection extends EventEmitter {
     this._requestHistoricalTelemetry();
     this._initiateSubscriptionTelemetry();
 
+    this.openmct.router.on('change:path', this._abortRequests);
```

Comment:
I'm curious about why `destroy` is not sufficient here? I would expect `destroy` to be called on any Telemetry Collections associated with a view that is being navigated away from.

Reply:
yeah it should be, I was seeing cases where other requests were hanging (in large display layouts) and though this may help in a race condition between navigation being called and the TC being destroyed. It may not be necessary.